File: csharp/basics/WordFrequency.cs
// Count the words of a fixed text, most frequent first and alphabetically within a tie.
// Run: dotnet run WordFrequency.cs (file-based app, .NET 10 or later), or as Program.cs in a console project

const string text = "the quick brown fox jumps over the lazy dog the fox barks";

var ranked = text
    .Split(' ', StringSplitOptions.RemoveEmptyEntries)
    .GroupBy(word => word)
    .OrderByDescending(group => group.Count())
    .ThenBy(group => group.Key, StringComparer.Ordinal);

foreach (var group in ranked)
{
    Console.WriteLine($"{group.Key} {group.Count()}");
}

File: kotlin/basics/WordFrequency.kt
// Count the words of a fixed text, most frequent first and alphabetically within a tie.
// Run: kotlinc WordFrequency.kt -include-runtime -d word_frequency.jar && java -jar word_frequency.jar

const val TEXT = "the quick brown fox jumps over the lazy dog the fox barks"

fun main() {
    TEXT.split(Regex("\\s+"))
        .groupingBy { it }
        .eachCount()
        .entries
        .sortedWith(compareByDescending<Map.Entry<String, Int>> { it.value }.thenBy { it.key })
        .forEach { (word, count) -> println("$word $count") }
}
